CRITICAL - 2022-06-10 01:55:45 --> Call to a member function findAll() on null
#0 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\vendor\codeigniter4\framework\system\CodeIgniter.php(832): App\Controllers\CV->index()
#1 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\vendor\codeigniter4\framework\system\CodeIgniter.php(419): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\CV))
#2 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\vendor\codeigniter4\framework\system\CodeIgniter.php(326): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#3 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\public\index.php(37): CodeIgniter\CodeIgniter->run()
#4 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(43): require_once('C:\\xampp\\htdocs...')
#5 {main}
CRITICAL - 2022-06-10 01:56:18 --> Invalid argument supplied for foreach()
#0 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\app\Views\CvView.php(184): CodeIgniter\Debug\Exceptions->errorHandler(2, 'Invalid argumen...', 'C:\\xampp\\htdocs...', 184, Array)
#1 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\vendor\codeigniter4\framework\system\View\View.php(210): include('C:\\xampp\\htdocs...')
#2 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\vendor\codeigniter4\framework\system\View\View.php(213): CodeIgniter\View\View->CodeIgniter\View\{closure}()
#3 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\vendor\codeigniter4\framework\system\Common.php(1089): CodeIgniter\View\View->render('CvView', Array, true)
#4 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\app\Controllers\CV.php(29): view('CvView')
#5 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\vendor\codeigniter4\framework\system\CodeIgniter.php(832): App\Controllers\CV->index()
#6 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\vendor\codeigniter4\framework\system\CodeIgniter.php(419): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\CV))
#7 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\vendor\codeigniter4\framework\system\CodeIgniter.php(326): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#8 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\public\index.php(37): CodeIgniter\CodeIgniter->run()
#9 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(43): require_once('C:\\xampp\\htdocs...')
#10 {main}
ERROR - 2022-06-10 02:30:13 --> Error connecting to the database: No connection could be made because the target machine actively refused it.

CRITICAL - 2022-06-10 02:30:13 --> Unable to connect to the database.
Main connection [MySQLi]: No connection could be made because the target machine actively refused it.

#0 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\vendor\codeigniter4\framework\system\Database\BaseConnection.php(577): CodeIgniter\Database\BaseConnection->initialize()
#1 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\app\Models\Report.php(13): CodeIgniter\Database\BaseConnection->query('SELECT SUM(tota...')
#2 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\app\Controllers\Dashboard.php(11): App\Models\Report->getMonthlyEarn()
#3 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\vendor\codeigniter4\framework\system\CodeIgniter.php(832): App\Controllers\Dashboard->index()
#4 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\vendor\codeigniter4\framework\system\CodeIgniter.php(419): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Dashboard))
#5 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\vendor\codeigniter4\framework\system\CodeIgniter.php(326): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#6 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\public\index.php(37): CodeIgniter\CodeIgniter->run()
#7 C:\xampp\htdocs\TB2-Perpustakaan-PWeb2\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(43): require_once('C:\\xampp\\htdocs...')
#8 {main}
